atlp-rwanda/atlp-devpulse-fn
# Application Skeleton Implementation

**Description**
This PR introduces skeleton loading components throughout the application to improve user experience by providing visual feedback during data fetching. These skeleton loaders mimic the structure of the final content, making the interface more intuitive and responsive while the content is loading.

**Description of Task Completed**

- Implemented skeleton loading components across key areas of the application.
- Skeleton loaders are designed to represent the final structure of the content, enhancing the visual feedback to users during loading times.
- Integrated loaders with existing features to ensure a smooth transition when content is being fetched from the backend.

**How to test this PR(locally)**

- Clone the repo and check out the branch ft-skeleton-implementation
- Set up environment variables as indicated in .env.example
- Run npm install followed by npm run dev
- Test loading states in following areas:
     - Navigate through different sections
     - Observe skeleton animations during data fetching
     - Verify smooth transitions between states

**Related Issue**
_https://github.com/atlp-rwanda/atlp-devpulse-fn/issues/161_ 

**Screenshots** 

![NavbarMembers](https://github.com/user-attachments/assets/0171ce3a-2881-4a5d-b0de-ec3d7232d310)

![traineeApplicant](https://github.com/user-attachments/assets/a23f2e8a-9b69-43bc-a1f2-f8c96b502673)

![editApplicantInfo](https://github.com/user-attachments/assets/3fce9f8a-ff51-4e50-96df-eb0910f1e544)

![trash](https://github.com/user-attachments/assets/ab152c33-07df-41ef-a475-3acd2c9c788a)
